### PR TITLE
Allow a speedup by getting the list of packages only once

### PR DESCRIPTION
**Pull request by lecorguille**
As you can observe [here](https://gitlab.com/ifb-elixirfr/cluster/r-packages/-/blob/master/vars/r-4.0.2-full.yml), we have a LOT of packages.

And since the default behaviour of the role is to fetch the package list at each R library, it take ages.

This PR will provide a switch to only get the list once.

### CODE REVIEW

**Comment by lecorguille** [comment]
https://github.com/Oefenweb/ansible-r/pull/51/checks?check_run_id=1961637784#step:5:8
```
  ansible-lint
  shell: /bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.9.1/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.1/x64/lib
A valid role namespace is required, edit meta/main.yml and assure that
author field is also a valid namespace:

galaxy_info:
  author: your_galaxy_namespace

See: https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespace-limitations
```

We hit this issue: https://github.com/ansible-community/ansible-lint/issues/1400
Which is fixed but not released yet: https://github.com/ansible-community/ansible-lint/pull/1401

**Comment by lecorguille** [comment]
The new version of ansible-lint still make the test fail.

https://github.com/ansible-community/ansible-lint/blob/master/src/ansiblelint/_prerun.py#L169

The role name must contain 2 letters

File: tasks/packages.yml [hunk 1]
```diff
@@ -8,19 +8,23 @@
...
+    - src: R-list-installed-packages.j2
+      dest: R-list-installed-packages
     - src: R-install-package.j2
       dest: R-install-package
     - src: R-remove-package.j2
       dest: R-remove-package
     - src: R-update-package.j2
       dest: R-update-package
-    - src: R-update-packages.j2
-      dest: R-update-packages
```
Comment (tersmitten) [inline tasks/packages.yml:18]: Why is this removed

File: tasks/packages.yml [hunk 2]
```diff
@@ -51,6 +57,7 @@
   changed_when: "_r_remove_package.stdout_lines[-1] is defined and _r_remove_package.stdout_lines[-1] == 'changed'"
   with_items: "{{ r_packages }}"
   when:
+    - r_list_packages_once == false or item.name in r_installed_packages
```
Comment (tersmitten) [inline tasks/packages.yml:60]: Can you fix the cs issues?

File: templates/usr/local/bin/R-install-package.j2 [hunk 3]
```diff
@@ -8,6 +8,7 @@ if (is.null(argv) | length(argv) < 1) {
 
 type = ifelse(is.na(argv[2]), '{{ r_packages_type }}', argv[2]);
 lib = ifelse(is.na(argv[3]), '{{ r_packages_lib }}', argv[3]);
+r_version = '{{ r_version }}';
```
Comment (tersmitten) [inline templates/usr/local/bin/R-install-package.j2:11]: What does this add?

File: templates/usr/local/bin/R-list-installed-packages.j2 [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+argv = commandArgs(trailingOnly = TRUE);
+if (is.null(argv) | length(argv) < 1) {
+  cat("Usage: R-list-installed-packages [lib]\n");
+  q(status = 1);
+}
+
+lib = ifelse(is.na(argv[1]), '{{ r_packages_lib }}', argv[1]);
+
+cat(paste(installed.packages(lib.loc = lib)[, 'Package'], collapse=","))
```
Comment (tersmitten) [inline templates/usr/local/bin/R-list-installed-packages.j2:11]: I think that newlines are easier to parse. And then you can use `stdout_lines` in the `task`. What do you think?

File: templates/usr/local/bin/R-install-package.j2 (hunk 3 above)
Comment (lecorguille) [reply templates/usr/local/bin/R-install-package.j2]: `r_version` is used below but never created.

File: tasks/packages.yml (hunk 1 above)
Comment (lecorguille) [reply tasks/packages.yml:18]: It isn't used anywhere in the role, but ok, I restored it.

File: templates/usr/local/bin/R-list-installed-packages.j2 (hunk 4 above)
Comment (lecorguille) [reply templates/usr/local/bin/R-list-installed-packages.j2]: Indeed. Done!

File: tasks/packages.yml (hunk 2 above)
Comment (lecorguille) [reply tasks/packages.yml]: Done

File: .ansible-lint [hunk 5]
```diff
@@ -1,3 +1,6 @@
 ---
 warn_list:
   - '106'
+
+skip_list:
+  - role-name
```
Comment (tersmitten) [inline .ansible-lint:6]: `106` is `role-name`

File: templates/R-install-package.j2 [hunk 6]
```diff
@@ -8,6 +8,7 @@ if (is.null(argv) | length(argv) < 1) {
 
 type = ifelse(is.na(argv[2]), '{{ r_packages_type }}', argv[2]);
 lib = ifelse(is.na(argv[3]), '{{ r_packages_lib }}', argv[3]);
+r_version = '{{ r_version }}';
```
Comment (tersmitten) [inline templates/R-install-package.j2:11]: No need to define this (it's a `jinja` `if`)

File: .ansible-lint (hunk 5 above)
Comment (lecorguille) [reply .ansible-lint]: Fix

File: templates/R-install-package.j2 (hunk 6 above)
Comment (lecorguille) [reply templates/R-install-package.j2]: Oups, indeed! 
Sorry for that
